aten/src/ATen/functorch/DynamicLayer.cpp

```diff
@@ -83,6 +83,18 @@ static DynmetaData& getGlobalDynmetaData() {
   return kDynMetaDataSingleton;
 }
 
+namespace {
+thread_local bool kIsAutogradFunctionAllowed = false;
```

Comment:
I would expect that if I enabled autograd.Function in one thread, all spawned threads would also have autograd.Function enabled (not certain if this will come up, but thinking that it might because of multithreading of the backward pass combined with higher order gradients). This version means that all threads will end up starting with the false behavior, right?

Reply:
You're right, ATen/ThreadLocalState needs to understand this. Thanks for catching